lib/iris/common/resolve.py

```diff
@@ -57,12 +64,260 @@
 
 
 class Resolve:
+    """
+    At present, :class:`~iris.common.resolve.Resolve` is used by Iris solely
+    during cube arithmetic to combine a left-hand :class:`~iris.cube.Cube`
+    operand and a right-hand :class:`~iris.cube.Cube` operand into a resultant
+    :class:`~iris.cube.Cube` with common metadata, suitably auto-transposed
+    dimensions, and an appropriate broadcast shape.
+
+    However, the capability and benefit provided by :class:`~iris.common.resolve.Resolve`
+    may be exercised as a general means to easily and consistently combine the metadata
+    of two :class:`~iris.cube.Cube` operands together into a single resultant
+    :class:`~iris.cube.Cube`. This is highlighted through the following use case
+    patterns.
+
+    Firstly, creating a ``resolver`` instance with *specific* :class:`~iris.cube.Cube`
+    operands, and then supplying ``data`` with suitable dimensionality and shape to
+    create the resultant resolved :class:`~iris.cube.Cube`, e.g.,
+
+    .. testsetup::
+
+        import iris
+        import numpy as np
+        from iris.common import Resolve
+        cube1 = iris.load_cube(iris.sample_data_path("A1B_north_america.nc"))
+        cube2 = iris.load_cube(iris.sample_data_path("E1_north_america.nc"))[0]
+        cube2.transpose()
+        cube3, cube4 = cube1, cube2
+        data = np.zeros(cube1.shape)
+        data1 = data * 10
+        data2 = data * 20
+        data3 = data * 30
```

Comment:
Note that, the `testsetup` instances created here for this doc-string is available for all other doc-strings, thanks to the magic of `sphinx`. 

Hence, there is no need to replicate the `testsetup` directive per doc-string test.